lib/driver.js

```diff
@@ -317,10 +317,11 @@ class AndroidDriver extends BaseDriver {
       let msg = 'The desired capabilities must include either an app, package or browser';
       log.errorAndThrow(msg);
     }
-    // make sure that the capabilities don't have both `app` and `browser`
+    // warn if the capabilities have both `app` and `browser, although this
+    // is common with selenium grid
     if (caps.browserName && caps.app) {
-      let msg = 'The desired capabilities should not include both an app and a browser';
-      log.errorAndThrow(msg);
+      let msg = 'The desired capabilities should generally not include both an app and a browser';
```

Comment:
Changed message to include the word "generally" since it's not necessarily bad, also make a note in the comment that it's common when using Appium with Selenium Grid.